src/planetary_system.jl

```diff
@@ -352,6 +352,340 @@ function generate_period_and_sizes_power_law(s::Star, sim_param::SimParam; num_p
+        Pc_lower, Pc_upper = P_lower/insert_pl_out_ρ, P_upper/insert_pl_in_ρ # Pc_lower should decrease if insert_pl_out_ρ > 1, and likewise Pc_upper should increase if insert_pl_in_ρ < 1
+        if verbose
+            println("Pc_blocked: ", (Pc_lower, Pc_upper))
+        end
+        push!(Pc_segments_unstable, (Pc_lower, Pc_upper))
+
+        pl_start += pl_per_cl[c]
+    end
+    Pc_segments_unstable = sort(Pc_segments_unstable)
+
+    Pc_segments_unstable_nonoverlapping = Tuple{Float64,Float64}[]
+    x_start, x_stop = Pc_segments_unstable[1]
+    for i in 1:length(Pc_segments_unstable)-1
+        if x_stop > Pc_segments_unstable[i+1][1] # segments overlap
+            x_stop = Pc_segments_unstable[i+1][2]
+        else # segments do not overlap
+            if Pc_segments_unstable[i+1][1]/x_stop > max_ratio
+                push!(Pc_segments_unstable_nonoverlapping, (x_start, x_stop))
+                x_start, x_stop = Pc_segments_unstable[i+1]
+            else # segments do not overlap, but there is not enough room between the two segments to insert the cluster
+                x_stop = Pc_segments_unstable[i+1][2]
+            end
+        end
+    end
+    push!(Pc_segments_unstable_nonoverlapping, (x_start, x_stop))
+    if verbose
+        println("Pc_blocked with no room: ", Pc_segments_unstable_nonoverlapping)
+    end
+    return Pc_segments_unstable_nonoverlapping
+end
+
+function compute_blocked_regions_non_overlapping(segments_blocked::Array{Tuple{Float64,Float64},1})
+    # NOTE: this function assumes that the input segments are sorted in increasing order!
+    # Make a list of non-overlapping segments:
+    segments_blocked_nonoverlapping = Tuple{Float64,Float64}[]
+    x_start, x_stop = segments_blocked[1]
+    for i in 1:length(segments_blocked)-1
+        if x_stop > segments_blocked[i+1][1] # segments overlap
+            x_stop = segments_blocked[i+1][2]
+        else # segments do not overlap
+            push!(segments_blocked_nonoverlapping, (x_start, x_stop))
+            x_start, x_stop = segments_blocked[i+1]
+        end
+    end
+    push!(segments_blocked_nonoverlapping, (x_start, x_stop))
+    return segments_blocked_nonoverlapping
+end
+
+function compute_allowed_regions_cdf_lognormal(segments_blocked::Array{Tuple{Float64,Float64},1}; μ::Float64=0., σ::Float64=1., x_min::Float64=0., x_max::Float64=Inf)
+    @assert 0 <= x_min < x_max <= Inf
+
+    # Make a list of non-overlapping segments in period deemed unstable:
+    segments_blocked_nonoverlapping = compute_blocked_regions_non_overlapping(segments_blocked)
+
+    # Make a list of non-overlapping, allowed segments in the cdf:
+    cdf_segments_allowed = Tuple{Float64,Float64}[]
+    n_seg_max = length(segments_blocked_nonoverlapping)+1 # maximum number of allowed segments if all the unstable segments fit between x_min and x_max
+    for i in 1:n_seg_max
+        if i==1
+            if x_min < segments_blocked_nonoverlapping[i][1]
```

Comment:
I think you could simplify to one if for xmin and one if for xmax.

Reply:
Done, although I had to change the "else" condition to "elseif 1 < i < n_seg_max".

Reply:
Good call.  
FWIW, I was thinking of replacing all of 550-563 with something like
```julia
cdf_start = ( i==1 && x_min < segments_blocked_nonoverlapping[i][1] ) ? x_min ; cdf_lognormal(segments_blocked_nonoverlapping[i-1][2]; μ=μ, σ=σ) )
cdf_stop =  ( i==n_seg_max && x_max > segments_blocked_nonoverlapping[i-1][2] ) ? x_max : cdf_lognormal(segments_blocked_nonoverlapping[i][1]; μ=μ, σ=σ)
push!(cdf_segments_allowed, (cdf_start, cdf_stop))
```

Reply:
I don't think that would work, since the i=1 and i=n_seg_max iterations should not push anything at all if they overlap with any of the segments.

Reply:
Ok, I see.